### PR TITLE
Enable enable_job_stack_trace by default

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]
Ok, but this is a query-level setting. Will it be enabled globally?

**Comment by yakov-olkhovskiy** [comment]
@alexey-milovidov it is set to all threads attaching a group - `ThreadStatus::applyGlobalSettings()` :
https://github.com/ClickHouse/ClickHouse/pull/51349/files#diff-f423985f33dd5d63d4755a441bdab81710a9104242ba01306c77cc7751ab7523

**Comment by yakov-olkhovskiy** [comment]
@alexey-milovidov please assess Stateless tests fail and Performance Comparison - I'm not exactly sure how serious it is and if caused by enabling this feature...

**Comment by Algunenano** [comment]
Please don't ignore CI: `Upgrade check (debug) — Changed settings are not reflected in the settings changes history (see changed_settings.txt)`

Reverting. Please fix and resend

File: src/Core/SettingsChangesHistory.cpp [hunk 1]
```diff
@@ -78,6 +78,7 @@ static std::initializer_list<std::pair<ClickHouseVersion, SettingsChangesHistory
     },
     {"24.10",
         {
+            {"enable_job_stack_trace", false, true, "Enable by default collecting stack traces from job's scheduling."},
```
Comment (yakov-olkhovskiy) [inline src/Core/SettingsChangesHistory.cpp:81]: @Algunenano I thought I did it here...

Comment (alexey-milovidov) [reply src/Core/SettingsChangesHistory.cpp:81]: Ok, but the next version is 24.11.

Comment (yakov-olkhovskiy) [reply src/Core/SettingsChangesHistory.cpp:81]: yes, already figure it out 🤔

**Pull request merged**